internal/app/utils_test.go

```diff
@@ -5,6 +5,71 @@ import (
 	"testing"
 )
 
+func TestReplaceAtIndex(t *testing.T) {
+	type args struct {
+		in string
+		r  rune
+		i  int
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "last_colon",
+			args: args{"abzd", 'c', 2},
+			want: "abcd",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := replaceAtIndex(tt.args.in, tt.args.r, tt.args.i)
+			if err != nil {
+				t.Errorf("replaceAtIndex() unexpected error: %v", err)
+			}
+			if got != tt.want {
+				t.Errorf("replaceAtIndex() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestOciRefToFilename(t *testing.T) {
+	tests := []struct {
+		name string
+		in   string
+		want string
+	}{
+		{
+			name: "no_repo",
+			in:   "my-chart:1.2.3",
+			want: "my-chart-1.2.3.tgz",
+		},
+		{
+			name: "two_colons",
+			in:   "my:chart:1.2.3",
+			want: "my:chart-1.2.3.tgz",
+		},
+		{
+			name: "with_Host",
+			in:   "my-repo.example.com/charts/my-chart:1.2.3",
```

Comment:
doesn't oci ref start with `oci://`? If so may adding a test case for this as well

Reply:
Sure, I can do that.

Reply:
done.